conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DialogueInterfaceGenerator.java

```diff
@@ -74,7 +80,44 @@ private JavaFile generate(
                 .map(endpoint -> apiMethod(endpoint, returnTypeMapper))
                 .collect(toList()));
 
-        serviceBuilder.addMethod(methodGenerator.generate(def));
+        MethodSpec staticFactoryMethod = methodGenerator.generate(def);
+        serviceBuilder.addMethod(staticFactoryMethod);
+
+        serviceBuilder.addMethod(MethodSpec.methodBuilder("of")
+                .addModifiers(Modifier.STATIC, Modifier.PUBLIC)
+                .addJavadoc(
+                        "Creates an asynchronous/non-blocking client for a $L service.",
+                        def.getServiceName().getName())
+                .returns(staticFactoryMethod.returnType)
+                .addParameter(Channel.class, StaticFactoryMethodGenerator.CHANNEL)
+                .addParameter(ConjureRuntime.class, StaticFactoryMethodGenerator.RUNTIME)
+                .addCode(CodeBlock.builder()
+                        .add(
+                                "if ($L instanceof $T) { return $L(($T) $L, $L); }\n",
+                                StaticFactoryMethodGenerator.CHANNEL,
+                                EndpointChannelFactory.class,
+                                staticFactoryMethod.name,
+                                EndpointChannelFactory.class,
+                                StaticFactoryMethodGenerator.CHANNEL,
+                                StaticFactoryMethodGenerator.RUNTIME)
```

Comment:
This isn't likely to produce issues like #908 because the factory is called once and the resulting EndpointChannel is reused for the lifespan of the client. :+1: